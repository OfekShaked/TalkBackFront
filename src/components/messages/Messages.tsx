import Message from './message/Message';

import './MessagesStyle.scss';

import AlwaysScrollToBottom from '../AlwaysScrollToBottom';

interface IMessages{
  messages:any,
  name:String|null
}

const Messages = ( props:IMessages ) => {

  return (
  <div className="messages">
    {props.messages.map((message:any, i:any) => <div key={i}><Message message={message} name={props.name} /></div>)}
    <AlwaysScrollToBottom/>
  </div>)
};

export default Messages;