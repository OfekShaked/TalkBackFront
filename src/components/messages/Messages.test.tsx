import React from 'react';
import renderer from 'react-test-renderer';
import Messages from './Messages';
import Message from './message/Message'
import Adapter from '@wojtekmaj/enzyme-adapter-react-17';
import { mount, configure } from 'enzyme';

configure({ adapter: new Adapter() });

const messages = [{text:"Hello there!",sender:"george",conversationId:"123"},{text:"Hello there!",sender:"george",conversationId:"123"},{text:"Hello there!",sender:"george",conversationId:"123"},{text:"Hello there!",sender:"george",conversationId:"123"},{text:"Hello there!",sender:"george",conversationId:"123"}]

describe('messages tests', () => {

    it('renders messages component', async () => {
        const component = renderer.create(
                <Messages name={"ofek"} messages={messages}></Messages>
        );
        let tree = component.toJSON();
        expect(tree).toMatchSnapshot();
    });

    it('renders all messages', async () => {
        window.HTMLElement.prototype.scrollIntoView = function() {}; //mocking a scrollIntoView cause its not implemented
        const component = mount(
            <Messages name={"ofek"} messages={messages}></Messages>
            );
        const messagesInDiv = component.find(Message);
        expect(messagesInDiv).toHaveLength(5);
    });

    it('has not messages',async()=>{
        window.HTMLElement.prototype.scrollIntoView = function() {}; //mocking a scrollIntoView cause its not implemented
        const component = mount(
            <Messages name={"ofek"} messages={[]}></Messages>
            );
        const messagesInDiv = component.find(Message);
        expect(messagesInDiv).toHaveLength(0);
    })
    

    

})

