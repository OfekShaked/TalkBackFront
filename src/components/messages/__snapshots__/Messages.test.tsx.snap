// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`messages tests renders messages component 1`] = `
<div
  className="messages"
>
  <div>
    <div
      className="messageContainer justifyStart"
    >
      <div
        className="messageBox backgroundLight"
      >
        <p
          className="messageText colorDark"
        >
          Hello there!
        </p>
      </div>
      <p
        className="sentText pl-10 "
      />
    </div>
  </div>
  <div>
    <div
      className="messageContainer justifyStart"
    >
      <div
        className="messageBox backgroundLight"
      >
        <p
          className="messageText colorDark"
        >
          Hello there!
        </p>
      </div>
      <p
        className="sentText pl-10 "
      />
    </div>
  </div>
  <div>
    <div
      className="messageContainer justifyStart"
    >
      <div
        className="messageBox backgroundLight"
      >
        <p
          className="messageText colorDark"
        >
          Hello there!
        </p>
      </div>
      <p
        className="sentText pl-10 "
      />
    </div>
  </div>
  <div>
    <div
      className="messageContainer justifyStart"
    >
      <div
        className="messageBox backgroundLight"
      >
        <p
          className="messageText colorDark"
        >
          Hello there!
        </p>
      </div>
      <p
        className="sentText pl-10 "
      />
    </div>
  </div>
  <div>
    <div
      className="messageContainer justifyStart"
    >
      <div
        className="messageBox backgroundLight"
      >
        <p
          className="messageText colorDark"
        >
          Hello there!
        </p>
      </div>
      <p
        className="sentText pl-10 "
      />
    </div>
  </div>
  <div />
</div>
`;
